sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -4136,6 +4140,52 @@ private static SqlQuerySpec createLogicalPartitionScanQuerySpec(
             invalidPartitionExceptionRetryPolicy);
     }
 
+    @Override
+    public Mono<List<PartitionKeyRange>> getPartitionKeyRanges(String collectionLink) {
+        InvalidPartitionExceptionRetryPolicy invalidPartitionExceptionRetryPolicy = new InvalidPartitionExceptionRetryPolicy(
+            this.collectionCache,
+            null,
+            collectionLink,
+            new HashMap<>());
+
+        RxDocumentServiceRequest request = RxDocumentServiceRequest.create(
+            this,
+            OperationType.Query,
+            ResourceType.Document,
+            collectionLink,
+            null);
+
+        invalidPartitionExceptionRetryPolicy.onBeforeSendRequest(request);
+
+        return ObservableHelper.inlineIfPossibleAsObs(
+            () -> getPartitionKeyRangeInternal(request, collectionLink),
+            invalidPartitionExceptionRetryPolicy);
+    }
+
+    private Mono<List<PartitionKeyRange>> getPartitionKeyRangeInternal(RxDocumentServiceRequest request,
```

Comment:
the logic in this one seems to be similar as getFeedRangesInternal?

Reply:
It is possible , but we wont be getting much benefit , infact extra cost of unboxing and boxing of  Utils.ValueHolder for PartitionKeyRange in getFeedRangesInternal